wfe/web-front-end.go

```diff
@@ -914,6 +1098,30 @@ func (wfe *WebFrontEndImpl) BuildID(response http.ResponseWriter, request *http.
 	response.WriteHeader(http.StatusOK)
 	detailsString := fmt.Sprintf("Boulder=(%s %s) Golang=(%s) BuildHost=(%s)", core.GetBuildID(), core.GetBuildTime(), runtime.Version(), core.GetBuildHost())
 	if _, err := fmt.Fprintln(response, detailsString); err != nil {
+		logEvent.Error = err.Error()
 		wfe.log.Warning(fmt.Sprintf("Could not write response: %s", err))
 	}
 }
+
+func (wfe *WebFrontEndImpl) logTerminatedRequest(logEvent requestEvent) {
```

Comment:
I'd like you to make this function a defer-only function:
- Rename to `logRequestIfError` (or similar)
- Move the `if logEvent.Error != nil` inside the method
   - This cleans up the defers above, so they're all one-line and less-subject to edits